test/hotspot/jtreg/compiler/intrinsics/TestLongDivMod.java

```diff
@@ -0,0 +1,126 @@
+* @run driver compiler.intrinsics.TestLongDivMod
+*/
+
+package compiler.intrinsics;
+import compiler.lib.ir_framework.*;
+import java.util.random.RandomGenerator;
+import java.util.random.RandomGeneratorFactory;
+
+
+public class TestLongDivMod {
+    private final int BUFFER_SIZE = 1024;
+    private long[] dividends = new long[BUFFER_SIZE];
+    private long[] divisors = new long[BUFFER_SIZE];
+    private long[] quotients =  new long[BUFFER_SIZE];
+    private long[] remainders =  new long[BUFFER_SIZE];
+    private RandomGenerator rng;
+
+    public static void main(String args[]) {
+        TestFramework.run(TestLongDivMod.class);
+    }
+
+    public TestLongDivMod() {
+        rng = RandomGeneratorFactory.getDefault().create(0);
+        for (int i = 0; i < BUFFER_SIZE; i++) {
+            dividends[i] = rng.nextLong();
+            divisors[i] = rng.nextLong();
+        }
+    }
+
+    @Test // needs to be run in (fast) debug mode
+    @Warmup(10000)
+    @IR(counts = {"UDivL", ">= 1"}) // Atleast one UDivL node is generated if intrinsic is used
+    public void testDivideUnsigned() {
+        for (int i = 0; i < BUFFER_SIZE; i++) {
+            quotients[i] = Long.divideUnsigned(dividends[i], divisors[i]);
+        }
+        checkResult("divideUnsigned");
+    }
+
+    @Test // needs to be run in (fast) debug mode
+    @Warmup(10000)
+    @IR(counts = {"UModL", ">= 1"}) // Atleast one UModL node is generated if intrinsic is used
+    public void testRemainderUnsigned() {
+        for (int i = 0; i < BUFFER_SIZE; i++) {
+            remainders[i] = Long.remainderUnsigned(dividends[i], divisors[i]);
+        }
+        checkResult("remainderUnsigned");
+    }
+
+
+    @Test // needs to be run in (fast) debug mode
+    @Warmup(10000)
+    @IR(counts = {"UDivModL", ">= 1"}) // Atleast one UDivModL node is generated if intrinsic is used
+    public void testDivModUnsigned() {
+        for (int i = 0; i < BUFFER_SIZE; i++)  divmod(dividends[i], divisors[i], i);
+        checkResult("divmodUnsigned");
+    }
+
+    private void divmod(long dividend, long divisor, int i) {
+        quotients[i] = Long.divideUnsigned(dividend, divisor);
+        remainders[i] = Long.remainderUnsigned(dividend, divisor);
+    }
+
+    public void checkResult(String mode) {
+        for (int i=0; i < BUFFER_SIZE; i++) {
+            long quo =  divideUnsigned(dividends[i], divisors[i]);
+            long rem = remainderUnsigned(dividends[i], divisors[i]);
+            boolean mismatch;
+            switch (mode) {
+                case "divideUnsigned": mismatch = (quotients[i] != quo); break;
+                case "remainderUnsigned": mismatch = (remainders[i] != rem); break;
+                case "divmodUnsigned": mismatch = (quotients[i] != quo || remainders[i] != rem); break;
+                default: throw new IllegalArgumentException("incorrect mode");
+            }
+            if (mismatch) {
+                throw new RuntimeException("Test failed");
```

Comment:
It would be good to print dividend, divisor, operation, actual result and expected result here.

Reply:
Please see the updated error message in the recent commit.